uqcsbot/scripts/bgg.py

```diff
@@ -0,0 +1,149 @@
+    if query.status_code != 200:
+        return None
+    result = fromstring(query.text)[0]
+    parameters: Dict[str, Any] = {}
+    parameters["categories"] = set()
+    parameters["mechanics"] = set()
+    parameters["subranks"] = {}
+    parameters["identity"] = identity
+
+    for i in result:
+        # sets the range of players
+        if i.tag == "poll" and i.attrib.get("name") == "suggested_numplayers":
+            players = set()
+            for j in i:
+                votes = 0
+                for k in j:
+                    votes += (int(k.attrib["numvotes"])
+                              * (-1 if k.attrib["value"] == "Not Recommended" else 1))
+                if votes > 0:
+                    players.add(j.attrib["numplayers"])
+            if players:
+                parameters["min_players"] = min(players)
+                parameters["max_players"] = max(players)
+        # sets the name of the board game
+        elif i.tag == "name" and i.attrib.get("type") == "primary":
+            parameters["name"] = i.attrib["value"]
+        # adds a category
+        elif i.tag == "link" and i.attrib.get("type") == "boardgamecategory":
+            parameters["categories"].add(i.attrib["value"])
+        # adds a mechanic
+        elif i.tag == "link" and i.attrib.get("type") == "boardgamemechanic":
+            parameters["mechanics"].add(i.attrib["value"])
+        # sets the user ratings
+        elif i.tag == "statistics":
+            for j in i[0]:
+                if j.tag == "average":
+                    parameters["score"] = j.attrib["value"]
+                if j.tag == "usersrated":
+                    parameters["users"] = j.attrib["value"]
+                if j.tag == "ranks":
+                    for k in j:
+                        if (k.attrib.get("name") == "boardgame"
+                                and k.attrib.get("value").isnumeric()):
+                            n = int(k.attrib.get("value"))
+                            o = "tsnrhtdd"[(n/10 % 10 != 1) * (n % 10 < 4) * n % 10::4]
+                            parameters["rank"] = f"{n:d}{o:s}"
+                        elif k.attrib.get("value").isnumeric():
+                            n = int(k.attrib.get("value"))
+                            o = "tsnrhtdd"[(n/10 % 10 != 1) * (n % 10 < 4) * n % 10::4]
+                            s = " ".join(k.attrib["friendlyname"].split(" ")[:-1])
+                            parameters["subranks"][s] = f"{n:d}{o:s}"
+        # sets the discription
+        elif i.tag == "description":
+            parameters["description"] = i.text
+        # sets the minimum playing time
+        elif i.tag == "minplaytime":
+            parameters["min_time"] = i.attrib["value"]
+        elif i.tag == "maxplaytime":
+            parameters["max_time"] = i.attrib["value"]
+
+    return parameters
+
+
+def format_board_game_parameters(parameters: dict) -> str:
+    message = (f"*{parameters.get('name', ':question:'):s}*\n"
```

Comment:
Format could be a bit fancier, e.g. link title to BGG, bold/italics, more emoji 🧑🕒⭐🎖⚙??